File: platform-ecom-backend/shared-commons/src/main/java/com/ecom/common/util/APIResponse.java
package com.ecom.common.util;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Standard API response wrapper.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class APIResponse<T> {
    private String message;
    private boolean success;
    private T data;
}

File: frontend-client/types/admin/api-response.ts
/**
 * Generic APIResponse wrapper from backend
 * All CRUD operations return this structure
 */
export interface APIResponse<T> {
    message: string;
    success: boolean;
    data: T;
}

/**
 * Helper to unwrap APIResponse and get the data
 * @throws Error if response status is false
 */
export function unwrapApiResponse<T>(response: APIResponse<T>): T {
    if (!response.success) {
        throw new Error(response.message || 'Operation failed');
    }
    return response.data;
}

/**
 * Extract success message from APIResponse
 */
export function getSuccessMessage<T>(response: APIResponse<T>): string {
    return response.message;
}
